File: src/main/java/org/homework20241215/shop/ShopBuilder.java
package org.homework20241215.shop;


import java.util.HashMap;
import java.util.Map;

public class ShopBuilder {

    private String name;
    private Map<Item, Integer> items = new HashMap<>();

    public ShopBuilder setName(String name) {
        this.name = name;
        return this;
    }

    public ShopBuilder addItem(Item item, int count) {
        this.items.put(item, count);
        return this;
    }

    public Shop build() {
        return new Shop(name, items);
    }

}

File: src/main/java/org/homework20241222/shop/ShopBuilder.kt
package org.homework20241222.shop

class ShopBuilder {

    private var name: String? = null
    private var items: Map<Item, Int> = HashMap()

    fun setName(name: String): ShopBuilder {
        this.name = name
        return this
    }

    fun addItem(item: Item, count: Int): ShopBuilder {
        this.items += item to count
        return this
    }

    fun build(): Shop {
        return Shop(name.toString(), items)
    }

}
